Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.js

```diff
@@ -67,24 +74,99 @@ gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawStar = function(x1, y1,
     this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
     this._graphics.moveTo(200,200);
     this._graphics.arcTo(x1, y1, x2, y2, radius);
-    //this._graphics.closePath();
+    if(closePath){
+        this._graphics.closePath();
+    }
     this._graphics.endFill();
 };
 */
 
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawBezierCurve = function(x1, y1, cpX, cpY, cpX2, cpY2, x2, y2) {
+    this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
+    this._graphics.moveTo(x1, y1);
+    this._graphics.bezierCurveTo(cpX, cpY, cpX2, cpY2, x2, y2);  
+    this._graphics.endFill();
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawQuadraticCurve = function(x1, y1, cpX, cpY, x2, y2) {
+    this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
+    this._graphics.moveTo(x1, y1);
+    this._graphics.quadraticCurveTo(cpX, cpY, x2, y2);
+    this._graphics.endFill();
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.beginFillPath = function() {
+    this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.endFillPath = function() {
+    this._graphics.endFill();
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathMoveTo = function(x1, y1) {
+    this._graphics.moveTo(x1, y1);
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathLineTo = function(x1, y1) {
+    if(this._graphics.graphicsData.length === 0){
+        this._graphics.moveTo(0, 0);
+    }
+    this._graphics.lineTo(x1, y1);
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathBezierCurveTo = function(cpX, cpY, cpX2, cpY2, toX, toY) {
+    if(this._graphics.graphicsData.length === 0){
+        this._graphics.moveTo(0, 0);
+    }
+    this._graphics.bezierCurveTo(cpX, cpY, cpX2, cpY2, toX, toY);  
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathArc = function(x1, y1, radius, startAngle, endAngle, anticlockwise) {
+    if(this._graphics.graphicsData.length === 0){
+        this._graphics.moveTo(0, 0);
+    }
+    this._graphics.arc(x1, y1, radius, gdjs.toRad(startAngle), gdjs.toRad(endAngle), anticlockwise ? true : false);
+};
+/*
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathArcTo = function(x1, y1, x2, y2, radius) {
+    this._graphics.arcTo(x1, y1, x2, y2, radius);
+};
+*/
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawPathQuadraticCurveTo = function(cpX, cpY, toX, toY) {
+    if(this._graphics.graphicsData.length === 0){
+        this._graphics.moveTo(0, 0);
+    }
+    this._graphics.quadraticCurveTo(cpX, cpY, toX, toY);  
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.closePath = function() {
+    this._graphics.closePath();  
+};
+
+/*
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.hasBegunFillPath = function() {
```

Comment:
If this is not necessary, let's remove this. In case you want to retrieve it, it will always be stored in previous commits in this PR, so it's never lost.

Reply:
ok, then i delete also the comment for ArcTo

Reply:
Yes, better keep the code clean. ArcTo can always be added later.

Reply:
Yes, better keep the code clean. ArcTo can always be added later.